docs/router.md

```diff
@@ -0,0 +1,204 @@
+                            method: "get",
+                            components: {
+                                middlewareOneA: {
+                                    type: "my.package.middlewareThatPassesThingsOn",
+                                    options: {
+                                        namespace: "middlewareOneA"
+                                    }
+                                },
+                                middlewareOneB: {
+                                    type: "my.package.middlewareThatResponds",
+                                    options: {
+                                        priority: "after:middlewareOneA"
+                                    }
+                                }
+                            }
+                        }
+                    },
+                    middlewareTwoB: {
+                        type: "my.package.middlewareThatResponds",
+                        options: {
+                            path:     "/foo"
+                            priority: "after:router1"
+                        }
+                    }
+                }
+            }
+        }
+    }
+});
+```
+
+Now, if a request for `GET /foo` is received, it will be evaluated against `router2`, which can work with it.  `router2`
+will route the request to `middlewareTwoA` and then `router1` in turn.  Although `router1` can handle the method, its
+path no longer matches.   The request will be routed to `middlewareTwoB`.
+
+If request for `GET /foo/foo` is received, it will be evaluated against `router2`, which can work with it.  `router2`
+will router the request to `middlewareTwoA` and then evaluate it against `router1`.  `router1` can handle both the
+method and the relative path, so it will be given the chance to route the request to `middlewareOneA` and
+`middlewareTwoB` in turn.
+
+If a request for `POST /foo/foo` is received, it will be evaluated against `router2`, which can work with any path that
+_begins with `/foo`_, and can work with the correct method.  `router2` will route the request to `middlewareTwoA` and
+then `router1` in turn.  Although `router1` can handle the specific path, it cannot handle the method.  The request will
+be routed to `middlewareTwoB`.
+
+# `gpii.express.router`
+
+An instance of [`gpii.express`](express.md) will automatically attempt to wire in anything with this gradeName into its
+routing table.  To be meaningfully useful, a router must have one or more child [middleware](middleware.md) or router
+components.
+
+## Component Options
+
+| Option          | Type                    | Description |
+| --------------- | ----------------------- | ----------- |
+| `method`        | `{String}`              | This grade provides the ability to limit itself to only operate on requests that match a particular [HTTP method](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html).  Support lowercased string values, such as `get`, `post`, `put`, or `delete`. |
```

Comment:
Just to illustrate the style I use for the future:

| Option              | Type       | Description |
| ------------------- | ---------- | ----------- |
| `requiredNoDefaults` | `{Number}` | A number that you have to set.  An error is thrown if this is not set. |
| `requiredWithDefaults`  |  `{Number}` | A number blah blah.  Set to `0` by default. |
| `optionalAlpha` (optional) | `{String}` | A string blah blah blah. |
| `optionalBeta` (optional) | `{String}` | A string blah blah blah. |

This is:

1. Required fields, alphabetically.
2. Optional fields, alphabetically, with "(optional)" after their `Option` value.
3. Defaults at the end of the `Description` value.